extension/src/prototypes/ui-components/TimelineBar.tsx

```diff
@@ -0,0 +1,324 @@
+import { styled, Slider, sliderClasses } from "@mui/material";
+import { useEffect, useMemo, useRef, useState } from "react";
+
+type TimelineBarProps = {
+  startDate: Date;
+  endDate: Date;
+  currentDate: Date;
+  timezone: string;
+  onChange: (event: Event, value: number | number[]) => void;
+};
+
+type Duration = {
+  msStart: number;
+  msEnd: number;
+  msDuration: number;
+};
+
+const defaultFullWidth = 248;
+const minTickWidth = 5;
+const endPadding = 30;
+const labelWidth = 60;
+
+const tickGroupSettings = [
+  { name: "second", ms: 1000 },
+  { name: "min", ms: 60000 },
+  { name: "tenmin", ms: 600000 },
+  { name: "halfhour", ms: 1800000 },
+  { name: "hour", ms: 3600000 },
+  { name: "day", ms: 86400000 },
+  { name: "month", ms: 2419200000 }, // 28 days, should not use this ms in tick calc since month is not fixed
+  { name: "year", ms: 31536000000 }, // 365 days, should not use this ms in tick calc since year is not fixed
+];
+
+const getTicks = (
+  duration: Duration,
+  fullWidth: number,
+  msStep: number,
+  name: string,
```

Comment:
IMO: It should be named `type`?
```suggestion
  type: "month" | "yeah",
```

Reply:
Yes you're right, i updated this with `type TickUnit = "second" | "min" | "tenmin" | "halfhour" | "hour" | "day" | "month" | "year";`